sympy/tensor/array/arrayop.py

```diff
@@ -279,6 +281,16 @@ def permutedims(expr, perm):
 
     # Get the inverse permutation:
     iperm = ~perm
+    new_shape = perm(expr.shape)
+
+    if isinstance(expr, SparseNDimArray):
+        new_array = {}
```

Comment:
call it `new_dict`, it's not an array.

Reply:
I was calling it `new_array` to respect the existing convention. Some lines below there is a list being called as `new_array` and then passed to the constructor. I think the code might be more comprehensible if we keep that convention.

Reply:
Which convention are you talking about? Suppose you were to write the code in C++, how would you call the variables?

Reply:
I was trying to use the same name for the same variable in one function, I thought it could help improve the coherence within the function. Anyway there is no longer this problem.

Reply:
No, it's confusing if you read the code. You are calling a dictionary an array, that's wrong.

Reply:
Ok, I got your point. I will keep this in mind. :-)